src/zabapgit_requirements.prog.abap

```diff
@@ -0,0 +1,201 @@
+      "! @raising lcx_exception | Cancelled by user or internal error
+      check_requirements IMPORTING it_requirements TYPE lcl_dot_abapgit=>ty_requirement_tab
+                                   iv_show_popup   TYPE abap_bool DEFAULT abap_true
+                         RAISING   lcx_exception,
+      "! Get a table with information about each requirement
+      "! @parameter it_requirements | Requirements
+      "! @parameter rt_status | Result
+      "! @raising lcx_exception | Internal error
+      get_requirement_met_status IMPORTING it_requirements  TYPE lcl_dot_abapgit=>ty_requirement_tab
+                                 RETURNING VALUE(rt_status) TYPE gty_status_tab
+                                 RAISING   lcx_exception.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    CLASS-METHODS:
+      show_requirement_popup IMPORTING it_requirements TYPE gty_status_tab
+                             RAISING   lcx_exception,
+      version_greater_or_equal IMPORTING is_status      TYPE gty_status
+                               RETURNING VALUE(rv_true) TYPE abap_bool.
+ENDCLASS.
+
+CLASS lcl_requirement_helper IMPLEMENTATION.
+  METHOD check_requirements.
+    DATA: lt_met_status TYPE gty_status_tab,
+          lv_answer     TYPE c LENGTH 1.
+
+    lt_met_status = get_requirement_met_status( it_requirements ).
+
+    IF iv_show_popup = abap_true.
+      show_requirement_popup( lt_met_status ).
+    ENDIF.
+
+    LOOP AT lt_met_status TRANSPORTING NO FIELDS WHERE met = abap_false.
+      EXIT.
+    ENDLOOP.
+
+    IF sy-subrc = 0.
+      CALL FUNCTION 'POPUP_TO_CONFIRM'
+        EXPORTING
+          text_question = 'The project has unmet requirements. Install anyways?'
+        IMPORTING
+          answer        = lv_answer.
+      IF lv_answer <> '1'.
+        lcx_exception=>raise( 'Cancelling because of unmet requirements.' ).
+      ENDIF.
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD get_requirement_met_status.
+    DATA: lt_installed TYPE STANDARD TABLE OF cvers_sdu.
+    FIELD-SYMBOLS: <ls_requirement>    TYPE lcl_dot_abapgit=>ty_requirement,
+                   <ls_status>         TYPE gty_status,
+                   <ls_installed_comp> TYPE cvers_sdu.
+
+    CALL FUNCTION 'DELIVERY_GET_INSTALLED_COMPS'
+      TABLES
+        tt_comptab       = lt_installed
+      EXCEPTIONS
+        no_release_found = 1
+        OTHERS           = 2.
+    IF sy-subrc <> 0.
+      lcx_exception=>raise( |Error from DELIVERY_GET_INSTALLED_COMPS { sy-subrc }| ) ##NO_TEXT.
+    ENDIF.
+
+    LOOP AT it_requirements ASSIGNING <ls_requirement>.
+      APPEND INITIAL LINE TO rt_status ASSIGNING <ls_status>.
+      <ls_status>-component = <ls_requirement>-component.
+      <ls_status>-required_release = <ls_requirement>-min_release.
+      <ls_status>-required_patch = <ls_requirement>-min_patch.
+
+      READ TABLE lt_installed WITH KEY component = <ls_requirement>-component
+                              ASSIGNING <ls_installed_comp>.
+      IF sy-subrc = 0.
```

Comment:
Then the component is not installed at all, which means <ls_status>-met will stay initial (which is `abap_false`).